### PR TITLE
fix(api): Fix miscellaneous bugs in deck CLI

### LINKED ISSUE #3986: After a factory reset, and before the first successful GUI pipette calibration, pipettes are way too low
# Overview

Without having already established good pipette calibrations through the pipette calibration GUI, pipettes will be positioned too low. They will be so low that they will crash into the z switch of the tip probe box during tip probing, preventing a successful calibration unless the tip is taken off and tip probing is faked by hitting the switches manually.

This occurs after CLI calibration on fresh SD cards. It also occurs after doing a tip length factory reset through the GUI.

We've seen this issue across multiple OT-2s.

# Reproducing

1. Feed an OT-2 a fresh SD card. We've seen this on a few different system versions, but the  VERSION.json corresponding to the sdcard.img I used is: `{"buildroot_version": "v1.0.3-1-gc4bf505b1", "buildroot_sha": "c4bf505b1562e9d9a7a3544a19cd16151a3b23f6", "buildroot_branch": "opentrons-develop", "buildroot_buildid": "opentrons-buildroot:61c73d8b-97f6-4bc9-a3b6-6ef4b482b286", "build_type": "dev", "opentrons_api_version": "3.11.3", "opentrons_api_sha": "b88c73ba9a79758cf130d2c24d8700aa5c13bcc8", "opentrons_api_branch": "edge", "update_server_version": "3.11.3", "update_server_sha": "b88c73ba9a79758cf130d2c24d8700aa5c13bcc8", "update_server_branch": "edge"}`
2. Do CLI calibration. The tip probe stage (`p`) may fail initially, but after multiple `p`+`spacebar` iterations with manual "helping," it will stabilize and succeed. See also #3985.
3. Reboot and upload a protocol through the GUI. I used one with two P300s. When you do pipette calibration, the pipettes will be too low, despite having done a successful tip probe in CLI calibration.
4. After many iterations with manual intervention, pipette calibration will eventually stabilize and succeed.
5. Do a **Reset** > **Tip Length**. Observe you're now back at step 3.

# Reading Material

I've attached a .zip with a history of JSON files.

  1. `robot_settings_0_after_cli_calibration.json` and `deck_calibration_0_after_cli_calibration`.json are from immediately after step 2, above.
  2. `robot_settings_1_after_much_pipette_calibration_finagling.json` is immediately after step 4, above.
  3. `robot_settings_2_after_factory_reset.json` is immediately after step 5, above.

[3986.zip](https://github.com/Opentrons/opentrons/files/3561724/3986.zip)

### LINKED ISSUE #4114: api: apiv1: cli deck cal gives incorrect results for z using non-p300 gen1 pipettes
## overview

CLI deck calibration in APIv1 does its own positioning. It determines the "height" of the deck to move to and compare to the saved points by finding the tip length. However, it does that _before_ deciding what tip length to use, so the Z output for pipettes other than p300 gen1 will be off by the difference between their default tip lengths and the default tip length for the p300 gen1.

## how to reproduce

do a CLI deck cal in apiv1 using something other than a p300 single and note that the z offset stored is incorrect - it should usually be around -23mm; if you do deck cal with a p10 this will be off by about 20mm. You can tell if you go to the deck in a protocol thereafter.

### PR DESCRIPTION
**Pull request by Laura-Danielle**
## overview

closes #3986 and closes #4114. This PR fixes some gross movement issues that happened in deck calibration on the api V1 system.

## changelog

- Calculate mount offset based on expected points with the corrected tip length of a particular pipette model
- Add the updated Z value from gantry calibration into the robot position tracker
- Move to points 1/2/3 with Z mount offset subtracted out

## review requests

Do CLI at least 2x in a row. Check that your data gets overwritten properly and that your tip probe center data is ~72-75.

### CODE REVIEW

File: api/src/opentrons/config/robot_configs.py [hunk 1]
```diff
@@ -413,6 +413,8 @@ def calculate_tip_probe_hotspots(
     nozzle_safe_z = round((size_z - tip_length)
                           + tip_probe_settings.z_clearance.normal, 3)
 
+    log.info(f"SIZE Z {size_z}")
```
Comment (sfoster1) [inline api/src/opentrons/config/robot_configs.py]: These probably shouldn't make it in

File: api/src/opentrons/deck_calibration/dc_main.py [hunk 2]
```diff
@@ -696,7 +719,7 @@ def main(loop=None):
     parser.add_argument(
         '-t', '--pickupTip',
         help='What to output during simulations',
-        default=None)
+        default=None, action='store_true')
```
Comment (sfoster1) [inline api/src/opentrons/deck_calibration/dc_main.py]: 🙌

File: api/src/opentrons/util/calibration_functions.py [hunk 3]
```diff
@@ -32,6 +32,7 @@ def probe_instrument(instrument, robot, tip_length=None) -> Point:
     safe_height = _calculate_safeheight(robot, tp.z_clearance.crossover)
 
     log.info("Moving to safe z: {}".format(safe_height))
+    log.info(f"Robot position info {robot.poses}")
```
Comment (sfoster1) [inline api/src/opentrons/util/calibration_functions.py]: Probably don't want this here, this ends up being an absolutely huge log message

File: api/src/opentrons/config/robot_configs.py (hunk 1 above)
Comment (Laura-Danielle) [reply api/src/opentrons/config/robot_configs.py]: whoops 🙈

**Pull request merged**